### PR TITLE
linter: check if bioconda recipe exists

### PR DESCRIPTION
**Pull request by dougalsutherland**
Re: https://github.com/bioconda/bioconda-recipes/issues/7574.

@bgruening, maybe you can suggest a more reliable will-always-be-in-bioconda recipe for the test? I picked `abawaca` just because it's near the top and the name looked fun. :)

### CODE REVIEW

**Comment by isuruf** [comment]
I don't think we want this to fail a PR. This should be a suggestion, not a failure. Related to https://github.com/conda-forge/conda-smithy/issues/628

**Comment by dougalsutherland** [comment]
Note that this check only runs on staged-recipes, so it’s not like a failure stops anything from happening or that it’s even annoying beyond the first notification. I don’t see that the red X is a huge deal for review, though maybe we should change the long message to something more transparently not-necessarily-terrible.

**Comment by jakirkham** [comment]
Something like this sounds like a good idea. We don't want to be breaking downstream unnecessarily.

Agree with @isuruf that it would be preferable to warn. That said, we could try working around the error issue with good documentation in `staged-recipes` and/or the webpage repo.

File: conda_smithy/tests/test_lint_recipe.py [hunk 1]
```diff
@@ -388,6 +388,18 @@ def test_maintainer_exists(self):
...
+        # Check that feedstock exists if staged_recipes
+        lints = linter.lintify({'package': {'name': 'abawaca'}}, recipe_dir='abawaca', conda_forge=True)
+        self.assertIn(expected_message, lints)
+        lints = linter.lintify({'package': {'name': 'abawaca'}}, recipe_dir='abawaca', conda_forge=False)
+        self.assertNotIn(expected_message, lints)
+        # No lint if in a feedstock
+        lints = linter.lintify({'package': {'name': 'abawaca'}}, recipe_dir='recipe', conda_forge=True)
+        self.assertNotIn(expected_message, lints)
+        lints = linter.lintify({'package': {'name': 'abawaca'}}, recipe_dir='recipe', conda_forge=False)
+        self.assertNotIn(expected_message, lints)
```
Comment (isuruf) [inline conda_smithy/tests/test_lint_recipe.py]: Can you move this block below line 416 and also add a test like in Line 404-416?

Comment (dougalsutherland) [reply conda_smithy/tests/test_lint_recipe.py]: Yep, looked at this block too quickly last night; fixing.

**Comment by mbargull** [comment]
> @bgruening, maybe you can suggest a more reliable will-always-be-in-bioconda recipe for the test? I picked abawaca just because it's near the top and the name looked fun. :)

Funny name indeed, but not really one of the "big ones" 😉. I'd rather use something like `samtools`.

Thanks for adding this lint check @dougalsutherland!

**Comment by bgruening** [comment]
@dougalsutherland let me know when you are in Germany at some point I own you a few beers :)
Do inter-organisation pings work? If so maybe we could add a ping to @bioconda/core ?

**Comment by dougalsutherland** [comment]
@bgruening I think inter-org pings just get silently dropped, but you can ping individual people from the team. And I'll be in a few neighboring countries in a few weeks, but not quite close enough. :)

**Comment by jakirkham** [comment]
Was thinking about suggesting an interorg ping, @bgruening, but I don't think they work. Maybe we could add a few select names instead? Alternatively we could add a team in conda-forge for bioconda reps and ping that team? Any of these (or something else) sound good?

**Comment by dougalsutherland** [comment]
I like the idea of a conda-forge/bioconda team to easily hit relevant people (and maybe vice-versa).

**Comment by jakirkham** [comment]
Sorry to dup, @dougalsutherland. Think you were a little faster at clicking. :)

**Comment by bgruening** [comment]
And so I did, thanks @isuruf!

File: conda_smithy/lint_recipe.py [hunk 2]
```diff
@@ -261,6 +261,14 @@ def run_conda_forge_lints(meta, recipe_dir, lints):
         if feedstock_exists:
             lints.append('Feedstock with the same name exists in conda-forge')
 
+        bio = gh.get_user('bioconda').get_repo('bioconda-recipes')
+        try:
+            bio.get_dir_contents('recipes/{}'.format(recipe_name))
+        except github.UnknownObjectException as e:
+            pass
+        else:
+            lints.append('Recipe with the same name exists in bioconda')
```
Comment (isuruf) [inline conda_smithy/lint_recipe.py]: Can you ping `@conda-forge/bioconda-recipes` here?

Comment (dougalsutherland) [reply conda_smithy/lint_recipe.py]: Sorry, I went on vacation and lost track of this. Updated now.

Comment (dougalsutherland) [reply conda_smithy/lint_recipe.py]: Sorry, I went on vacation and lost track of this. Updated (and rebased to latest `master`) now.

**Comment by dougalsutherland** [comment]
Not sure why coveralls thinks the entire `run_conda_forge_lints` function [is untested](https://coveralls.io/builds/15698759/source?filename=conda_smithy%2Flint_recipe.py#L271); it should be....

**Comment by CJ-Wright** [comment]
Something about at GH key missing I think.

**Comment by CJ-Wright** [comment]
If we remove the decorator from the `test_conda_forge_lints` the CI tests don't pass.

**Pull request merged**